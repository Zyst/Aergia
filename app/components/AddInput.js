// @flow
import React from "react";
import { connect } from "react-redux";
import { addTask } from "../actions/task";
import type { actionType } from "../reducers/task";

type Props = {
  cancel: () => void,
  dispatch: (action: actionType) => void
};

/* eslint-disable import/no-mutable-exports */
let AddInput = ({ cancel, dispatch }: Props) => {
  let task: ?HTMLInputElement;

  return (
    <form
      onSubmit={e => {
        e.preventDefault();

        if (task && task.value.trim()) {
          dispatch(addTask(task.value.trim(), 1));

          task.value = "";

          cancel();
        }
      }}
    >
      {/* eslint-disable jsx-a11y/no-autofocus */}
      <input
        ref={node => {
          task = node;
        }}
        type="text"
        autoFocus
        placeholder="Task name"
      />;
    </form>
  );
};
AddInput = connect()(AddInput);

export default AddInput;
