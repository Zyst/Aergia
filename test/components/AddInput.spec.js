import { spy } from "sinon";
import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import * as configure from "../../app/store/configureStore.prod";
import AddInput from "../../app/components/AddInput";

function setup() {
  const actions = {
    cancel: spy()
  };
  const store = configure.configureStore();

  const component = shallow(
    <Provider store={store}>
      <AddInput cancel={actions.cancel} />
    </Provider>
  );

  return {
    component,
    actions,
    store
  };
}

describe("Add Input component", () => {
  it("should be defined", () => {
    expect(AddInput).toBeDefined();
  });

  it("should match snapshot", () => {
    const { actions, store } = setup();

    const tree = renderer
      .create(
        <Provider store={store}>
          <AddInput cancel={actions.cancel} />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
