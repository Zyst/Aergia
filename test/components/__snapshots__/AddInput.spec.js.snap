// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Input component should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    placeholder="Task name"
    type="text"
  />
  ;
</form>
`;
